packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -3370,6 +3398,33 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
     _setSelection(newSelection, cause);
   }
 
+  /// Select a paragraph around the location of the last tap down.
+  ///
+  /// {@macro flutter.rendering.RenderEditable.selectPosition}
+  void selectParagraph({ required SelectionChangedCause cause }) {
+    selectParagraphAt(from: _lastTapDownPosition!, cause: cause);
+  }
+
+  /// Selects the paragraph at a given offset
+  ///
+  /// {@macro flutter.rendering.RenderEditable.selectPosition}
+  void selectParagraphAt({ required Offset from, required SelectionChangedCause cause }) {
+    assert(from != null);
+    _layoutText(minWidth: constraints.minWidth, maxWidth: constraints.maxWidth);
+    final TextPosition fromPosition = _textPainter.getPositionForOffset(globalToLocal(from - _paintOffset));
+    final TextRange boundary = _getParagraphBoundary(text!, fromPosition);
```

Comment:
Not sure if `_getParagraphBoundary` belongs in TextPainter. Regardless, I've implemented in this file for now. I noticed there is a `_textPainer.getLineBoundary` but that corresponds to the lines as they are rendered.